BlueDB/src/main/java/org/bluedb/api/BlueCollection.java

```diff
@@ -9,25 +9,88 @@
 import org.bluedb.api.keys.BlueKey;
 import org.bluedb.api.keys.ValueKey;
 
+/**
+ * A BlueCollection is a persisted map of keys (of type BlueKey) to values of object type T.
+ * 
+ * A collection has a name (to distinguish between collections in a BlueDb instance), a key type, and a value type.
+ * @param <T> object type of values to be serialized into collection
+ */
 public interface BlueCollection<T extends Serializable> {
 
+	/**
+	 * Creates (or returns existing) BlueIndex that maps objects of type keyType to values in the collection.
+	 * @param name index (one index per name per each collection)
+	 * @param keyType the type of each key which is used to lookup a value using the index (this must match the keyType of any existing index with the same name)
+	 * @param keyExtractor a function that maps a value to the keys by which the value should be indexed
+	 * @return a BlueIndex (existing index if it exists, otherwise a newly created index)
+	 * @throws BlueDbException if index doesn't exist or has a different keyType
+	 */
 	public <I extends ValueKey> BlueIndex<I, T> createIndex(String name, Class<I> keyType, KeyExtractor<I, T> keyExtractor) throws BlueDbException;
 
+	/**
+	 * Returns existing BlueIndex that maps objects of type keyType to values in the collection.

```

Comment:
Returns existing BlueIndex that maps objects of type ValueKey (UUIDKey, StringKey, IntegerKey, LongKey) to values in the collection